core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -746,6 +746,13 @@ public void failShard(String reason, Throwable e) {
         engine().failEngine(reason, e);
     }
 
+    /**
+     * Fails the shard without marking it as corrupted
+     */
+    public void softFailShard(String reason, Throwable e) {
```

Comment:
can we just make it an overload of failShard with a boolean, like in the engine? . I don't know what soft means :)

Also if we do that, let's change failShard(string reason, throwable e) to just call failShard(reason, true, e) ...